### DISCUSSION TITLE
Is there an oEmbed plugin?

### DISCUSSION THREAD

**Discussion by boris-glumpler** [Milkdown/milkdown#1329] [Q&A]
I was looking for a plugin that embeds content, like YouTube videos or tweets, when a URL is added on its own line and surrounded by empty lines, but couldn't find anything, unfortunately.

There are several remark plugins, like this one here for example: https://github.com/remark-embedder/transformer-oembed

I reckon I could do this fairly easily with some custom directives for select services, like `::youtube {#01ab2cd3efg}`, as there's already a good iframe example on https://milkdown.dev, but it seems to me that a general oEmbed plugin could be more useful. I'm not much of a JS/TS dev, tho, so I'd appreciate any pointers. Cheers!

**Comment by Saul-Mirone** [comment]
I think there is no one working on this. Do you want to create a community plugin? I'm willing to use that in the website playground.

**Reply by boris-glumpler** [reply]
We actually decided to implement directives for now as this gives us more options when displaying the embeds, like adding a confirmation box to load content from external sites. Having said that, I would still like to come back to adding a general oEmbed plugin once I get the Milkdown-FilamentPHP editor I'm working on atm published. And yes, once (if?) I get it working I'll release it as a community plugin!
